packages/connect-node-conformance/src/transport.ts

```diff
@@ -0,0 +1,79 @@
+import { ClientCompatRequest } from "./gen/connectrpc/conformance/v1alpha1/client_compat_pb.js";
+import {
+  Codec,
+  HTTPVersion,
+  Protocol,
+  Compression as ConformanceCompression,
+} from "./gen/connectrpc/conformance/v1alpha1/config_pb.js";
+import {
+  createConnectTransport,
+  createGrpcTransport,
+  compressionGzip,
+  compressionBrotli,
+  createGrpcWebTransport,
+} from "@connectrpc/connect-node";
+import { Compression } from "@connectrpc/connect/protocol";
+
+export function createTransport(req: ClientCompatRequest) {
+  let scheme = "http://";
+  if (req.serverTlsCert.length > 0) {
+    scheme = "https://";
+  }
+  const baseUrl = `${scheme}${req.host}:${req.port}`;
+
+  let httpVersion: "1.1" | "2";
+  switch (req.httpVersion) {
+    case HTTPVersion.HTTP_VERSION_1:
+      httpVersion = "1.1";
+      break;
+    case HTTPVersion.HTTP_VERSION_2:
+      httpVersion = "2";
+      break;
+    case HTTPVersion.HTTP_VERSION_3:
+      throw new Error("HTTP/3 is not supported");
+    default:
+      throw new Error("Unknown HTTP version");
+  }
+
+  let sendCompression: Compression | undefined = undefined;
+
+  switch (req.compression) {
+    case ConformanceCompression.GZIP:
+      sendCompression = compressionGzip;
+      break;
+    case ConformanceCompression.BR:
+      sendCompression = compressionBrotli;
+      break;
+    case ConformanceCompression.DEFLATE:
+    case ConformanceCompression.SNAPPY:
+    case ConformanceCompression.ZSTD:
+      throw new Error("Unsupported compression");
+    case ConformanceCompression.UNSPECIFIED:
+    case ConformanceCompression.IDENTITY:
+      break;
+  }
+
+  const sharedOptions = {
+    baseUrl,
+    httpVersion,
+    useBinaryFormat: req.codec === Codec.PROTO,
+    sendCompression,
+  };
```

Comment:
Since we're using `Any`, we have to provide a type registry in the JSON options here.